src/IECoreScene/Primitive.cpp

```diff
@@ -78,10 +81,27 @@ Imath::Box3f Primitive::bound() const
 		if( p )
 		{
 			const vector<V3f> &pp = p->readable();
-			for( size_t i=0; i<pp.size(); i++ )
-			{
-				result.extendBy( pp[i] );
-			}
+			using RangeType = tbb::blocked_range< const V3f* >;
+			tbb::this_task_arena::isolate( [ & result, & pp ]{
+				result = tbb::parallel_reduce( RangeType( pp.data(), pp.data() + pp.size(), 1000 ), Imath::Box3f(),
+				[]( const RangeType& range, const Imath::Box3f& value ) -> Imath::Box3f
+				{
+					Imath::Box3f b( value );
+					for( const V3f
+						*      it    = range.begin(),
+						*const itEnd = range.end(); it != itEnd; ++it )
+					{
+						b.extendBy( *it );
+					}
+					return b;
+				},
+				[]( const Imath::Box3f& lhs, const Imath::Box3f& rhs ) -> Imath::Box3f
+				{
+					Imath::Box3f b( lhs );
+					b.extendBy( rhs );
+					return b;
+				} );
+			} );
```

Comment:
We need to pass a `task_group_context` argument constructed from `task_group_context::isolated` here. Otherwise TBB cancellation for our tasks can be triggered by cancellation in unrelated tasks, causing us to silently return a partially complete result. There's a fairly detailed explanation of this in this Gaffer commit : https://github.com/GafferHQ/gaffer/commit/c5acf86f7b7afe90cd81082deb6ad51244be6c1e.

Reply:
done